### ISSUE TITLE
Home assistant addon

### ISSUE THREAD

**Issue by stplatt** [roflcoopter/viseron#576]
It'd be great for the home assistant community to have viseron available as a home assistant addon.

https://community.home-assistant.io/t/viseron-v2-2-0-self-hosted-local-only-nvr-and-ai-computer-vision-software/223152/47
https://community.home-assistant.io/t/viseron-v2-2-0-self-hosted-local-only-nvr-and-ai-computer-vision-software/223152/48
https://community.home-assistant.io/t/viseron-v2-2-0-self-hosted-local-only-nvr-and-ai-computer-vision-software/223152/60

**Comment by roflcoopter** [comment]
Indeed, sorry i havent done this yet. Hard to find the time for everything!

**Comment by MistressPlague** [comment]
+1 <3

**Comment by xeberth** [comment]
Vote up!

**Comment by yangminglong** [comment]
[WebRTC](https://github.com/AlexxIT/WebRTC) And [go2RTC](https://github.com/AlexxIT/go2rtc) It seems that there is good support for Home Assistant. 
if possible with [Viseron](https://github.com/roflcoopter/viseron) Combining them will be very perfect

**Comment by roflcoopter** [comment]
Viseron has support for WebRTC now.

I understand if many of you got tried of waiting but i will try to make an addon in the coming month or two

**Comment by Wolfensteinor** [comment]
Can't wait 😁😁😁😁

**Comment by roflcoopter** [comment]
Started working on this now finally. I have been holding off since I don't have a machine to run Home assistant OS on, but I just realized that Home Assistant provides a devcontainer to test locally, silly me.

Need to iron out some kinks regarding MJPEG streaming and subpath issues, but I can probably have a working first version out next week.

**Comment by roflcoopter** [comment]
I think you should be able to test now with https://github.com/roflcoopter/viseron-hass-app

It is pointing to the Docker tag `hass-app-test` atm which is the latest 3.5.0 release with the added changes in #1305 

When any of you get the time, can you test it out and see if it works?
MJPEG streaming does not work yet as mentioned previously

**Comment by roflcoopter** [comment]
MJPEG streaming should work now as well.

Anyone had the chance to test it out?

**Comment by john-** [comment]
I have not yet tested but can if that helps.  I am running dev / 6388aa8.  I can just install the add on with the instructions you provided and point it at my existing Viseron instance running in Docker?  Or do I need a special version of Viseron?

Update: I am running Home Assistant Core which does not have access to add-ons (as far as I can find).  Therefore, I will not be able to test.

**Comment by Wolfensteinor** [comment]
I tried it but I think this app creates its own viseron instance.
No way to point it at my current instance running on a separate machine and get notifications or something like that. I could be wrong. I'm new here 😅

**Comment by roflcoopter** [comment]
Yes the point of the addon is to run the Docker container under the Home Assistant supervisor.
I did see that some other addons provide a proxy type version that lets you connect to an instance outside of the supervisor, but the only benefit you get from that is that you can utilize Home Assistant Ingress if i understand it correctly.

You can always copy the config folder over to the new instance running in Home Assistant, but file paths will likely have changed so playback wont really work. I dont see an easy way of creating a migration path either for that so if you want to use the addon instead of the standalone container you might just have to bite the bullet and start over with recordings

**Comment by roflcoopter** [comment]
Sadly didnt get much testing on this but th PR is merged to `dev` and will be in the next release.
My own internal tests worked fine so we will see how it works out in the wild
